static/scripts/app-register.js

```diff
@@ -1,7 +1,55 @@
 const formbtn = document.querySelector('.btn');
+const formInputs = document.querySelectorAll('.form-inputs');
 // http://localhost:2228/register
 // const inputs = document.querySelectorAll('.form-inputs');
 
+function keyDownEvents(){
+    const REGEXP = regExpDelivery();
+
+    for(let i = 0; i < formInputs.length; i++){
+
+        formInputs[i].addEventListener('input', () => {
+            
+            let user_err = document.createElement('small');
+            if(!REGEXP[i].test(formInputs[i].value)){
+                // console.log(formInputs[i].nextElementSibling.nodeName)
+                if(formInputs[i].nextElementSibling?.classList.contains('user-err')){
+                    formInputs[i].nextElementSibling.remove();
+                }
+                user_err.classList.add('user-err');
+                user_err.innerText = keydownMessages(formInputs[i])[i];
+                formInputs[i].style.border = '2px solid #b90909'
+                formInputs[i].after(user_err);
+                formbtn.disabled = true
+                
+            } else if(REGEXP[i].test(formInputs[i].value)){
+                if(formInputs[i].nextElementSibling?.classList.contains('user-err')){
+                    formInputs[i].nextElementSibling.remove();
+                }
+                    user_err.remove();
+                    formInputs[i].style.border = '2px solid #dddddd';
+                    formInputs[i].addEventListener('focus', () => formInputs[i].style.border = '2px solid #0941b9');
+                    formInputs[i].addEventListener('blur', () => formInputs[i].style.border = '2px solid #dddddd')
+                    formbtn.disabled = false
+            }
+        })
+    }
+    
+    
+}
+
+keyDownEvents()
+
+function keydownMessages(inputName){
+    let arr = [];
```

Comment:
Why is it an array and not a map (object) where keys are names of the elements?  Like
`{
"username": "message for username"
}`

Reply:
coz it's simpler for me to do this kind of stuff with Arrays []

> Why is it an array and not a map (object) where keys are names of the elements? Like `{ "username": "message for username" }`

Reply:
Your call, I did not see this pattern much, hence the comment. You create a function that simply returns you an array and you always know the order of elements there. In addition, you know what page item maps to what item of array. To me it's a pure map use case.

Reply:
alright alright, lemme try this w/ Map and then we'll see

> Your call, I did not see this pattern much, hence the comment. You create a function that simply returns you an array and you always know the order of elements there. In addition, you know what page item maps to what item of array. To me it's a pure map use case.